git/repo/base.py

```diff
@@ -682,10 +682,12 @@ def blame_incremental(self, rev, file, **kwargs):
         data = self.git.blame(rev, '--', file, p=True, incremental=True, stdout_as_string=False, **kwargs)
         commits = dict()
 
-        stream = iter(data.splitlines())
+        stream = iter(data.split(b'\n'))
         while True:
             line = next(stream)  # when exhausted, casues a StopIteration, terminating this function
-
+            if line.strip() == '' or line.strip() == b'':
+                # Skip over empty lines
+                continue
```

Comment:
Ohh, this actually has nothing to do with the ^M chars in the fixture. It's because of this:

```python
>>> s = 'foo\nbar\n'
>>> s.splitlines()
['foo', 'bar']
>>> s.split('\n')
['foo', 'bar', '']
               ^^
```